api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -379,31 +379,32 @@ const (
 	PruneRetentionPolicy RetentionPolicy = "Prune"
 )
 
-// +kubebuilder:validation:Enum=ImageRegistry;openshift-samples
-type OptionalCapability string
-
-const ImageRegistryCapability OptionalCapability = OptionalCapability(configv1.ClusterVersionCapabilityImageRegistry)
-
-const OpenShiftSamplesCapability OptionalCapability = OptionalCapability(configv1.ClusterVersionCapabilityOpenShiftSamples)
-
-// capabilities allows disabling optional components at install time.
+// capabilities allows enabling or disabling optional components at install time.
+// When this is not supplied, the cluster will use the DefaultCapabilitySet defined for the respective
+// OpenShift version, minus the baremetal capability.
 // Once set, it cannot be changed.
+//
+// +kubebuilder:validation:XValidation:rule="has(self.enabled) && has(self.disabled) ? self.enabled.all(e, !(e in self.disabled)) : true", message="Capabilities can not be both enabled and disabled at once."
 type Capabilities struct {
-	// disabled when specified, sets the cluster version baselineCapabilitySet to None
-	// and sets all additionalEnabledCapabilities BUT the ones supplied in disabled.
-	// This effectively disables that capability on the hosted cluster.
-	//
-	// When this is not supplied, the cluster will use the DefaultCapabilitySet defined for the respective
-	// OpenShift version.
+	// enabled when specified, explicitly enables the specified capabilitíes on the hosted cluster.
+	// Once set, this field cannot be changed.
 	//
+	// +listType=atomic
+	// +immutable
+	// +optional
+	// +kubebuilder:validation:MaxItems=25
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Enabled is immutable. Changes might result in unpredictable and disruptive behavior."
+	Enabled []configv1.ClusterVersionCapability `json:"enabled,omitempty"`
+
+	// disabled when specified, explicitly disables the specified capabilitíes on the hosted cluster.
 	// Once set, this field cannot be changed.
 	//
 	// +listType=atomic
 	// +immutable
 	// +optional
 	// +kubebuilder:validation:MaxItems=25
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Disabled is immutable. Changes might result in unpredictable and disruptive behavior."
-	Disabled []OptionalCapability `json:"disabled,omitempty"`
+	Disabled []configv1.ClusterVersionCapability `json:"disabled,omitempty"`
```

Comment:
we don't want to consume configv1 here since that drops our enum validation. We want to be in control and gradually expose what can be enable or disable with the enum validation

Reply:
Same for Enabled

Reply:
Done